barcode_scanner.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+# Module:        barcode_scanner.py
+# Author:        Raks Raja - Development Team
+# Description:   This module is used to process the barcodes in the S3 bucket and updates the loaction IDs and the status of item line IDs
+# Copyrights:    2020 (c) All Rights Reserved
+
+# Python libraries
+import requests
+import itertools
+import re
+import json
+import random
+import boto3
+from boto3.session import Session
+
+from constant import Constant #Inheriting Constant class
+
+# S3 bucket connection params
+session = Session()
+s3 = session.resource('s3')
+s3_bucket = s3.Bucket(Constant.S3_BUCKET_NAME)
+
+
+class BarcodeScanner:
+
+    def update_item_line_status(item_line_id, location_id):
+        """Scans the item line and updates the location id and checked in, checked out status """
+        try:
+            check_in_check_out_url = Constant.BASE_URL + 'v2/inventory_item_lines/%s' % (item_line_id)
+            location_url = Constant.BASE_URL + 'v2/inventory_item_lines/%s' % (item_line_id)
+            location_definition_request = requests.get(location_url, headers=Constant.headers)
+            if location_definition_request.status_code != 200:
+                print('An exception has occured while processing the API request for the item line ID%s: %s %s' % (item_line_id, location_definition_request.status_code, location_definition_request.reason))
+                return
+            location_json = location_definition_request.json()
+            previous_location_id = location_json['location']['id']
+            check_in_check_out_status_finder_url = Constant.BASE_URL + 'v2/inventory_item_lines?line_id=%s' % (item_line_id)
+            check_in_check_out_status_definition_request = requests.get(check_in_check_out_status_finder_url, headers=Constant.headers)
+            if check_in_check_out_status_definition_request.status_code != 200:
+                print('An exception has occured while processing the API request for the item line ID %s: %s %s' % (item_line_id, check_in_check_out_status_definition_request.status_code, check_in_check_out_status_definition_request.reason))
+                return
+            check_in_check_out_status_response = check_in_check_out_status_definition_request.json()
+            if location_id == previous_location_id:
+                if check_in_check_out_status_response['checked_out']:
+                    value = False
+                    status = 'checked-in'
+                else:
+                    value = True
+                    status = 'checked-out'
+            else:
+                value = False
+                status = 'checked-in'
+            param = json.dumps({
+                "checked_out": value,
+                "location_id": location_id
+            })
```

Comment:
let's change this logic slightly based on the new `mode` param:
1. If `mode = 'checkin'`, set `checked_out: false` and `location_id: location_id`.
2. If `mode = 'checkout'` and `location_id == previous_location_id`, set `checked_out: true`.
3. If `mode = 'checkout'` and `location_id != previous_location_id`, do nothing.